Formula/vineyard.rb

```diff
@@ -0,0 +1,104 @@
+class Vineyard < Formula
+  include Language::Python::Virtualenv
+
+  desc "In-memory immutable data manager. (Project under CNCF)"
+  homepage "https://v6d.io"
+  url "https://github.com/v6d-io/v6d/releases/download/v0.6.0/v6d-0.6.0.tar.gz"
+  sha256 "26a024bd143a19f1ba9fb87fa9eb9fec53f19836d8214b35bc472fa2132033c3"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "apache-arrow"
+  depends_on "boost"
+  depends_on "etcd"
+  depends_on "etcd-cpp-apiv3"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "libgrape-lite"
+  depends_on "llvm"
+  depends_on "nlohmann-json"
+  depends_on "open-mpi"
+  depends_on "openssl@1.1"
+  depends_on "tbb"
+
+  fails_with gcc: "5"
+
+  def install
+    # make libclang.so/libclang.dll findable by clang.cindex.Index.create()
+    ENV["LD_LIBRARY_PATH"] = "#{ENV["LD_LIBRARY_PATH"]}:#{Formula["llvm"].opt_lib}"
+
+    # Ensure we're using the right version of Python
+    python = Formula["python@3.10"].opt_bin
+    ENV.prepend_path "PATH", python
+    # We need `libclang` which is packaged with `llvm`
+    ENV.prepend_path "PYTHONPATH", Formula["llvm"].opt_prefix/Language::Python.site_packages(python)
+
+    # link against system libc++ instead of llvm provided libc++
+    ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
+
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_CXX_STANDARD=14",
+                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
+                    "-DPYTHON_EXECUTABLE=#{python}/python",
```

Comment:
```suggestion
    python = Formula["python@3.10"].opt_bin/"python3"
    ENV.prepend_path "PATH", python.dirname
    # We need `libclang` which is packaged with `llvm`
    ENV.prepend_path "PYTHONPATH", Formula["llvm"].opt_prefix/Language::Python.site_packages(python)

    # link against system libc++ instead of llvm provided libc++
    ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib

    system "cmake", "-S", ".", "-B", "build",
                    "-DCMAKE_CXX_STANDARD=14",
                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
                    "-DPYTHON_EXECUTABLE=#{python}",
```